### PR TITLE
Fix/2519 facet heading

### PR DESCRIPTION
**Pull request by cy**
screen shot 2016-06-21 at 4 31 20 pm

### CODE REVIEW

**Comment by alex-wilmer** [comment]
lgtm 👍

File: app/scripts/components/facets/facets.directives.ts [hunk 1]
```diff
@@ -7,24 +7,56 @@ module ngApp.components.facets.directives {
...
       controller: "termsCtrl as tc",
       link: ($scope: IFacetScope, elem: ng.IAugmentedJQuery, attr: ng.IAttributes, ctrl: ITermsController) => {
+
+        $scope.collapsed = angular.isDefined($scope.collapsed) ? $scope.collapsed : false;
+        //after directive has been rendered, check if text overflowed then add tooltip
+        $timeout(() => {
+          const label = document.getElementById(`${$scope.title.toLowerCase().replace(/\s+/g, '-')}-facet-label`);
+          if (label && label.offsetWidth < label.scrollWidth) {
+            label.setAttribute('uib-tooltip', $scope.title);
+            $compile(label)($scope)
```
Comment (alex-wilmer) [inline app/scripts/components/facets/facets.directives.ts]: `;`

File: app/scripts/components/facets/facets.directives.ts [hunk 2]
```diff
@@ -56,11 +88,16 @@ module ngApp.components.facets.directives {
...
+        removable: "=",
       },
       replace: true,
       templateUrl: "components/facets/templates/facets-free-text.html",
-      controller: "freeTextCtrl as ftc"
+      controller: "freeTextCtrl as ftc",
+      link: ($scope: IFacetScope, elem: ng.IAugmentedJQuery, attr: ng.IAttributes, ctrl: IFreeTextController) => {
+        $scope.clear = () => {
+          ctrl.actives.forEach(term => ctrl.FacetService.removeTerm($scope.field, term));
+        }
```
Comment (alex-wilmer) [inline app/scripts/components/facets/facets.directives.ts]: `;` I hate these guys, but we're using'em!

File: app/scripts/components/facets/facets.directives.ts [hunk 3]
```diff
@@ -78,7 +115,12 @@ module ngApp.components.facets.directives {
       },
       replace: true,
       templateUrl: "components/facets/templates/facets-prefix.html",
-      controller: "freeTextCtrl as ftc"
+      controller: "freeTextCtrl as ftc",
+      link: ($scope: IFacetScope, elem: ng.IAugmentedJQuery, attr: ng.IAttributes, ctrl: IFreeTextController) => {
+        $scope.clear = () => {
+          ctrl.actives.forEach(term => ctrl.FacetService.removeTerm($scope.field, term));
+        }
```
Comment (alex-wilmer) [inline app/scripts/components/facets/facets.directives.ts]: `;`

File: app/scripts/components/facets/facets.directives.ts [hunk 4]
```diff
@@ -89,13 +131,20 @@ module ngApp.components.facets.directives {
...
       replace: true,
       templateUrl: "components/facets/templates/facets-date.html",
-      controller: "dateFacetCtrl as dfc"
+      controller: "dateFacetCtrl as dfc",
+      link: ($scope: IFacetScope, elem: ng.IAugmentedJQuery, attr: ng.IAttributes, ctrl: IDateFacetController) => {
+        $scope.collapsed = angular.isDefined($scope.collapsed) ? $scope.collapsed : false;
+        $scope.clear = () => {
+          ctrl.FacetService.removeTerm(ctrl.name, ctrl.$window.moment($scope.date).format('YYYY-MM-DD'));
+          ctrl.facetAdded = false;
+        }
```
Comment (alex-wilmer) [inline app/scripts/components/facets/facets.directives.ts]: `;`

File: app/scripts/components/facets/facets.directives.ts [hunk 5]
```diff
@@ -104,17 +153,26 @@ module ngApp.components.facets.directives {
...
-      controller: "rangeFacetCtrl as rfc"
+      controller: "rangeFacetCtrl as rfc",
+      link: ($scope: IFacetScope, elem: ng.IAugmentedJQuery, attr: ng.IAttributes, ctrl: IRangeFacetController) => {
+
+        $scope.collapsed = angular.isDefined($scope.collapsed) ? $scope.collapsed : false;
+        $scope.clear = () => {
+          ctrl.FacetService.removeTerm($scope.field, null, ">=");
+          ctrl.FacetService.removeTerm($scope.field, null, "<=");
+          ctrl.upperFacetAdded = ctrl.lowerFacetAdded = false;
+        }
```
Comment (alex-wilmer) [inline app/scripts/components/facets/facets.directives.ts]: `;` See wouldn't it just be nicer to not have these ^_^

Comment (cy) [inline app/scripts/components/facets/facets.directives.ts]: orz

**Comment by alex-wilmer** [comment]
This can be merged!

**Pull request merged**